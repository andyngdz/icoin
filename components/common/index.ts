export { ICoinLogo } from 'components/common/Logo'
export { Route } from 'components/common/Route'
