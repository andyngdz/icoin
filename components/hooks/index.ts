export * from './useChartData'
export * from './useGlobal'
export * from './useLivePrice'
export * from './useLivePrices'
export * from './usePagination'
export * from './useTheme'
export * from './useTime'
