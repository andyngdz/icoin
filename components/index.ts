export * from './asset-detail/AssetDetail'
export * from './asset-detail/AssetDetailContent'
export * from './asset-detail/AssetHighLow'
export * from './asset-detail/AssetSummary'
export * from './assets/AssetItem'
export * from './assets/AssetNameAndDescription'
export * from './assets/Assets'
export * from './assets/AssetsTable'
export * from './common/container-wrapper/ContainerWrapper'
export * from './common/logo/Logo'
export * from './common/pagination/Pagination'
export * from './common/paper-wrapper/PaperWrapper'
export * from './common/route/Route'
export * from './common/value-color/ValueColor'
export * from './disclaimer/Disclaimer'
export * from './exchanges/ExchangeContent'
export * from './exchanges/ExchangeItem'
export * from './exchanges/ExchangesAll'
export * from './exchanges/ExchangesAsset'
export * from './exchanges/ExchangeStatus'
export * from './header/Header'
export * from './header/Menu'
export * from './hooks'
export * from './line-chart/LineChart'
export * from './live-prices/LivePrices'
export * from './search/Search'
export * from './symbol-icon/SymbolIcon'
export * from './time-selection/TimeSelection'
export * from './top-bar/TopBar'
export * from './top-bar/TopBarContent'
