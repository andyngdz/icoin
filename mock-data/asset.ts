const asset = {
  changePercent24Hr: 3.1752942454749208,
  explorer: 'https://blockchain.info/',
  id: 'bitcoin',
  marketCapUsd: 649529734588.947600205682695,
  maxSupply: 21000000.0,
  name: 'Bitcoin',
  priceUsd: 34888.9719149353737679,
  rank: 1,
  supply: 18617050.0,
  symbol: 'BTC',
  volumeUsd24Hr: 14661390048.4857838209011476,
  vwap24Hr: 34081.3126994825452706
}

export { asset }
