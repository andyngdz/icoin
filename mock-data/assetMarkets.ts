const assetMarkets = {
  pageInfo: { hasNextPage: true, __typename: 'PageInfo' },
  edges: [
    {
      node: {
        id: '3a6e1109-9f90-4659-8985-5c8b9e36328b',
        exchangeId: 'binance',
        exchangeName: 'Binance',
        baseSymbol: 'BTC',
        quoteSymbol: 'USDT',
        volumeUsd24Hr: 3822582086.935666004007041,
        priceUsd: 47483.608071927724242,
        percentVolume: 18.2682261132248922,
        updatedAt: '2021-02-15T08:48:27.353Z',
        __typename: 'AssetMarket'
      },
      __typename: 'AssetMarketEdge'
    },
    {
      node: {
        id: '04a633d2-d43f-4b73-9e0c-22fc9195af8a',
        exchangeId: 'bit-z',
        exchangeName: 'Bit-Z',
        baseSymbol: 'BTC',
        quoteSymbol: 'USDT',
        volumeUsd24Hr: 1543652315.1777949383004728,
        priceUsd: 47489.0307680033636913,
        percentVolume: 7.3771573487587625,
        updatedAt: '2021-02-15T08:48:12.995Z',
        __typename: 'AssetMarket'
      },
      __typename: 'AssetMarketEdge'
    },
    {
      node: {
        id: '00ae5690-a93a-41aa-917f-6fc4b6e03b56',
        exchangeId: 'huobi',
        exchangeName: 'Huobi',
        baseSymbol: 'BTC',
        quoteSymbol: 'USDT',
        volumeUsd24Hr: 1466373888.025307072992641,
        priceUsd: 47504.9586871037659115,
        percentVolume: 7.0078415960059594,
        updatedAt: '2021-02-15T08:35:29.911Z',
        __typename: 'AssetMarket'
      },
      __typename: 'AssetMarketEdge'
    }
  ],
  __typename: 'AssetMarketConnection'
}

export { assetMarkets }
