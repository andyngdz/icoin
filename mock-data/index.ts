export * from './asset'
export * from './assetMarkets'
export * from './assetNode'
export * from './assets'
export * from './assetsString'
export * from './exchanges'
export * from './global-data'
export * from './histories'
export * from './historyString'
export * from './node'
