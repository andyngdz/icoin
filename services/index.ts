export * from './paginate'
export * from './chart'
export * from './format'
export * from './api'
export * from './cdn'
export * from './test-utils'
export * from './price-direction'
export * from './price'
export * from './routes'
