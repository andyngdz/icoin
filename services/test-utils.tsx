import { Provider } from 'react-redux'
import { render } from '@testing-library/react'
import { ThemeProvider } from '@material-ui/core'
import React from 'react'
import Store from 'states'
import Theme from 'styles'

const Providers = ({ children }) => {
  return (
    <Provider store={Store}>
      <ThemeProvider theme={Theme}>{children}</ThemeProvider>
    </Provider>
  )
}

const customRender = (ui, options = {}) =>
  render(ui, { wrapper: Providers, ...options })

export * from '@testing-library/react'

export { customRender as render }
