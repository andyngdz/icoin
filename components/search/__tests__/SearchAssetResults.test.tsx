import { customRender } from 'services/test-utils'
import { SearchAssetResults } from 'components'
import { search } from 'mock-data'

describe('SearchAssetResults', () => {
  it('Should render it correctly if assets are not empty', async () => {
    const renderResult = customRender(
      <SearchAssetResults assets={search.assets} />
    )
    expect(renderResult).toMatchSnapshot()
  })

  it('Should render it correctly if assets are empty', async () => {
    const renderResult = customRender(
      <SearchAssetResults
        assets={{
          edges: []
        }}
      />
    )
    expect(renderResult).toMatchSnapshot()
  })
})
