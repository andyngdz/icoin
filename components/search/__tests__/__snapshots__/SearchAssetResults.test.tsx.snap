// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchAssetResults Should render it correctly if assets are empty 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="MuiListItem-root MuiListItem-gutters"
      >
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2"
          >
            Your search did not match any documents.
          </h6>
          <span
            class="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary"
          >
            Make sure that all words are spelled correctly
          </span>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="MuiListItem-root MuiListItem-gutters"
    >
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2"
        >
          Your search did not match any documents.
        </h6>
        <span
          class="MuiTypography-root MuiTypography-caption MuiTypography-colorTextSecondary"
        >
          Make sure that all words are spelled correctly
        </span>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`SearchAssetResults Should render it correctly if assets are not empty 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        role="button"
        tabindex="0"
      >
        <div
          class="SearchAssetResults-icon-1"
        />
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <p
            class="MuiTypography-root MuiTypography-button"
          >
            Bitcoin
          </p>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            BTC
          </span>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        role="button"
        tabindex="0"
      >
        <div
          class="SearchAssetResults-icon-1"
        />
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <p
            class="MuiTypography-root MuiTypography-button"
          >
            Binance Coin
          </p>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            BNB
          </span>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        role="button"
        tabindex="0"
      >
        <div
          class="SearchAssetResults-icon-1"
        />
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <p
            class="MuiTypography-root MuiTypography-button"
          >
            Bitcoin Cash
          </p>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            BCH
          </span>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        role="button"
        tabindex="0"
      >
        <div
          class="SearchAssetResults-icon-1"
        />
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <p
            class="MuiTypography-root MuiTypography-button"
          >
            Bitcoin SV
          </p>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            BSV
          </span>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
      <div
        aria-disabled="false"
        class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
        role="button"
        tabindex="0"
      >
        <div
          class="SearchAssetResults-icon-1"
        />
        <div
          class="MuiListItemText-root MuiListItemText-multiline"
        >
          <p
            class="MuiTypography-root MuiTypography-button"
          >
            Binance USD
          </p>
          <span
            class="MuiTypography-root MuiTypography-caption"
          >
            BUSD
          </span>
        </div>
        <span
          class="MuiTouchRipple-root"
        />
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="SearchAssetResults-icon-1"
      />
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <p
          class="MuiTypography-root MuiTypography-button"
        >
          Bitcoin
        </p>
        <span
          class="MuiTypography-root MuiTypography-caption"
        >
          BTC
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="SearchAssetResults-icon-1"
      />
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <p
          class="MuiTypography-root MuiTypography-button"
        >
          Binance Coin
        </p>
        <span
          class="MuiTypography-root MuiTypography-caption"
        >
          BNB
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="SearchAssetResults-icon-1"
      />
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <p
          class="MuiTypography-root MuiTypography-button"
        >
          Bitcoin Cash
        </p>
        <span
          class="MuiTypography-root MuiTypography-caption"
        >
          BCH
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="SearchAssetResults-icon-1"
      />
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <p
          class="MuiTypography-root MuiTypography-button"
        >
          Bitcoin SV
        </p>
        <span
          class="MuiTypography-root MuiTypography-caption"
        >
          BSV
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
    <div
      aria-disabled="false"
      class="MuiButtonBase-root MuiListItem-root MuiListItem-gutters MuiListItem-button"
      role="button"
      tabindex="0"
    >
      <div
        class="SearchAssetResults-icon-1"
      />
      <div
        class="MuiListItemText-root MuiListItemText-multiline"
      >
        <p
          class="MuiTypography-root MuiTypography-button"
        >
          Binance USD
        </p>
        <span
          class="MuiTypography-root MuiTypography-caption"
        >
          BUSD
        </span>
      </div>
      <span
        class="MuiTouchRipple-root"
      />
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
